Add empty-states.mdx and update nav (#63)
primer/design: content/ui-patterns/empty-states.mdx

---
title: Empty states
---

import {Flex, Box, Text, BorderBox} from '@primer/components'

> Empty states appear when there is no content on a page or feature. The blankslate component is used to represent large empty states across our products. These guidelines demonstrate best practices for using the blankslate component and designing empty states. If you're looking for guidelines on implementation, please refer to [Primer CSS](https://primer.style/css/components/blankslate).

Reviewer: Can we try framing the "why" we use empty states here? The first two sentences feel duplicative. Perhaps just a slight tweak on wording (please edit as you see fit):
"Empty states are used to fill spaces when no content has been added yet, or is temporarily empty due to the nature of the feature. These guidelines demonstrate..."

Author: Yes! Sorry missed this in the proofread. Thank you for correcting it.

## The blankslate component
The blankslate is made up of several elements that work together to inform the user about a feature and how to proceed forward. Below are the different elements of the component and how to modify them.

![blankslate anatomy](https://user-images.githubusercontent.com/6846673/62806713-78795380-baa8-11e9-9cbd-1a56fef3247b.png)

### Graphic
The graphic can bring delight, preview an interface element, or represent the goal of the feature. Graphics should be placed intentionally and with thought about the intention of the content. Graphics also differ in meaning and appeal to the user, which is why the blankslate component has multiple variations. These variations are outlined later on this page.

### Primary Text

> Primary text is the first piece of information in the empty state presented to the user. It can make a user feel much more comfortable to engage with the content or begin a feature flow. Primary text should sound welcoming, human, and convey the intention of the feature.

Reviewer: The goal of this content should be to help someone make a decision about how to effectively use an empty state, the first 2 sentences don't help me make a decision. I don't think this content is bad, but I do think it could be more helpful. I'd suggest jumping straight into usage guidelines, using command/imperative tense (because this helps give people direction and answers). 

For example: "Use Primary text to make the purpose of the empty state clear, help users feel comfortable about the feature they're engaging with, and the flow that might follow...

Author: Noted, thank you!

### Secondary text
This optional text is used to inform the user about the feature in more detail. Secondary text should be brief and non-redundant if possible. From the text, users should  be able to understand the general purpose of the feature and how it may help them accomplish a goal.

### Primary action
Blankslates can and are encouraged to use one primary action. This button should lead to a feature or component creation flow. Button copy should be keep brief and descriptive. If a button requires further specification, consider adding an Octicon.

### Secondary action
Secondary actions are optional and are represented by a text link located below the primary action button. A secondary action is used to direct a user to additional content about the feature. This might look like "[Learn more about X](#)" or "[Check out the guide on X](#)" or simply "[Learn more](#)".

### Border

> The border is visible by default and helps define the structure of the blankslate component within the context of the page.

Reviewer: I thought the border was optional and invisible by default? This description could be more helpful, along the lines of "use the border to give structural boundaries when needed in context with other UI elements on the page..."

Author: Yes, thank you for catching this. Updated on the next commit.

> The border is invisible by default, but can be added to help define the structure of the blankslate component when needed. This can be particularly helpful in page layouts where the blankslate is not the only content on the screen.

Reviewer 2: The only part that feels like it's missing here is a link to the implementation guidelines. As these are design guidelines it's fine that we're not showing _how_ to add a border, but it would be useful to link to the part of the (future) implementation docs that show the border variant from here.

Reviewer 3: @ZainKho that link is https://primer.style/css/components/blankslate#add-border

Author: Will add!

> The border is invisible by default, but can be added to help define the structure of the blankslate component when needed. This can be particularly helpful in page layouts where the blankslate is not the only content on the screen. For implmenetation, check out the [Primer CSS blankslate component](https://primer.style/css/components/blankslate#add-border).

Reviewer 3: ```suggestion
The border is invisible by default, but can be added to help define the structure of the blankslate component when needed. This can be particularly helpful in page layouts where the blankslate is not the only content on the screen. For implementation, check out the [Primer CSS blankslate component](https://primer.style/css/components/blankslate#add-border).
```

## Variations
Empty states have multiple variations that can be used in different contexts.

### Pictograms

> [Pictograms](https://ghicons.github.com/) present a great preview of what should be in place of the empty state. Blankslates should default to using a pictogram for graphic elements.

Reviewer 3: This first sentence doesn't make sense to me. I might try rewording this to something like, "Pictograms can be used to represent the feature where the blankslate is found."

![pictogram blankslate](https://user-images.githubusercontent.com/6846673/62806619-39e39900-baa8-11e9-9de0-5b2f511d63da.png)

### Illustrations

> Illustrations should be used **sparingly** in blankslates - only in cases where it is necessary to evoke emotion from the user. For example, illustrations in a new user experience can help introduce octocats to the user and bring a more personalized experience.

Reviewer 3: Also, I don’t know why "for example" is here. My understanding is that the illustrations are used to continue to tie the octocat as a brand element of GitHub in order to add more personality to a specific experience. 
```suggestion
Illustrations should be used **sparingly** in blankslates. For example, illustrations in a new user experience can help introduce octocats to the user and bring a more personalized experience.
```

Author: It's just one example that sets the tone very specifically for where illustrations might be used since pictograms are our default. It also ties down to the first time user experience guidelines later on the page.

![illustration blankslate](https://user-images.githubusercontent.com/6846673/62806617-394b0280-baa8-11e9-9abc-ccbb76a9f63f.png)

### Compact

> Compact states are a great way to engage and educate a user within Organizations, Settings, and other nested navigation pages. They should replace content that is left-aligned. While the border is optional for all empty states, the Compact variation requires it to better support page structure.

Reviewer 3: I'm not sure if it's necessary to list out specific areas in the product for compact blankslates. I think it provides more understanding and flexibility if we write the guideline to show that compact blankslates  are use when there are more UI elements and the area is much more condensed.

Reviewer 3: Also I feel a little unclear on 
> They should replace content that is left-aligned.
The way this is written it feels as if all of the left-aligned content will turn into a blankslate.

Author: Ok, I can reword this.

> ![compact blankslate](https://user-images.githubusercontent.com/6846673/63041086-dbd5fd80-be7b-11e9-8929-3a9c72f53edc.png)

Reviewer 3: I might show the compact blankslate with and without the border or just without the border since that is the default.

## Content and copy

> Empty states should explain features _in addition_ to conveying intention. Below is a good example of conveying intention in the primary text and using the secondary text to inform.

Reviewer 3: ```suggestion
Empty states should explain features _in addition_ to conveying intention. In the example below, the primary text is used to provide a welcome invitation to creating a positive community while the secondary text supports this intent by informing the user on how to do this by providing a code of conduct.
```

> Voice and tone are important to making sure users are not only engaged, but also informed about features. Empty states should explain features _in addition_ to conveying intention. Below is a good example of conveying intention in the primary text and using the secondary text to inform.

Reviewer 3: ```suggestion
Empty states should explain features _in addition_ to conveying intention. Below is a good example of conveying intention in the primary text and using the secondary text to inform.
```

![code of conduct illustration blankslate](https://user-images.githubusercontent.com/6846673/62806616-394b0280-baa8-11e9-8a5e-45f0c51aaa22.png)

> ## New user experiences

Reviewer: Should this be "first time user experiences"? "New user" suggests "new to GitHub", whereas "first time experience" works if they are either new to GitHub or using a specific feature for the first time.

> For new user experiences, use illustration blankslates to playfully engage the user and introduce the Octocat as a symbol of GitHub. Primary text should welcome the user to the platform and feature. Secondary text should seek to educate the user, but at a simpler, less-technical level.

Reviewer 3: ```suggestion
For new user experiences, we suggest using illustrations for the graphic element of these blankslates to differentiate from the more common default blankslate.

Primary text should welcome the user to the platform and feature. Secondary text should seek to educate the user, but at a simpler, less-technical level.
```

> ![new user blankslate](https://user-images.githubusercontent.com/6846673/62813307-efb9e200-babe-11e9-93e7-a6fd45d7f942.png)

Reviewer 3: I'd update the alt tag to reflect "first time user blankslsate"
